.github/workflows/python-ci.yml

```
@@ -10,18 +14,28 @@ jobs:
         uses: actions/checkout@v3
         with:
           submodules: true
+      - name: Setup Python
+        uses: actions/setup-python@v4
+      - name: Load pip cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip
+          restore-keys: ${{ runner.os }}-pip
+      - name: Install linters
+        run: python -m pip install --upgrade black pylint
       - name: Black
-        uses: psf/black@stable
+        run: black . --check
       - name: Pylint
-        uses: cclauss/GitHub-Action-for-pylint@0.7.0
-        with:
-          args: pip install .; pylint src/**/*.py
+        run: pylint src/**/*.py
 
   test:
+    needs: lint
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```

Python 3.11 was released in October 
```suggestion
        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
```